### PR TITLE
RI-8219: Integration tests for array read endpoints

### PR DESCRIPTION
**Pull request by pawelangelow**
# What

Add integration coverage for the seven Array **read** endpoints landed
in #6064, plus a Redis 8.8-gated section in `keys/get-info` that exercises the
new `GetArrayKeyInfoResponse` `oneOf` branch.

Eight test files:

- `array/POST-databases-id-array-get_range.test.ts`      — ARGETRANGE
- `array/POST-databases-id-array-scan.test.ts`           — ARSCAN
- `array/POST-databases-id-array-get_length.test.ts`     — ARLEN
- `array/POST-databases-id-array-get_count.test.ts`      — ARCOUNT
- `array/POST-databases-id-array-get_next_index.test.ts` — ARNEXT
- `array/POST-databases-id-array-get_element.test.ts`    — ARGET
- `array/POST-databases-id-array-get_elements.test.ts`   — ARMGET
- `keys/POST-databases-id-keys-get_info.test.ts`         — Array branch

What the suite pins, beyond what unit specs already cover:

- decimal-string contract for `length` / `count` / scan-`index` / `next-index`
  (incl. a u64 boundary case above `Number.MAX_SAFE_INTEGER`);
- gap-preserving semantics — empty slots surface as JSON `null`;
- `RedisStringToBufferTransformer` null passthrough on `encoding=buffer`;
- explicit `limit: null` on scan treated as omitted;
- the 400 paths hardened in #6064 (reversed range, range > 1M, ARMGET
  `@ArrayMinSize(1)` and the per-call indexes cap, non-decimal / out-of-u64);
- `WrongType`, missing key, missing instance per endpoint;
- per-command ACL denials (`-arget`, `-arlen`, `-arcount`, `-arnext`,
  `-armget`, `-argetrange`, `-arscan`).

All assertions go through the existing `deps` / `validateApiCall` /
`getMainCheckFn` harness; every Array describe is gated by
`requirements('rte.version>=8.8')` so non-preview RTE jobs skip cleanly
rather than fail.

# Testing

- `yarn lint:api` — clean
- `yarn --cwd redisinsight/api type-check` — passes against the baseline
- `yarn --cwd redisinsight/api test:api` — runs the new suite end-to-end
  against an Array-capable RTE; skips on RTE jobs without Redis 8.8 preview.

---

> [!NOTE]
> **Low Risk**
> Test-only changes with no production API or runtime behavior modified; risk is limited to CI duration on Redis 8.8 array RTE jobs.
> 
> **Overview**
> Adds **Redis 8.8–gated** API integration coverage for the seven array **read** routes (`get-count`, `get-element`, `get-elements`, `get-length`, `get-next-index`, `get-range`, `scan`) using the existing `validateApiCall` / `getMainCheckFn` harness. Suites assert wire contracts that unit tests do not fully exercise end-to-end: **decimal-string** `length`/`count`/indexes (including u64 above `MAX_SAFE_INTEGER`), **sparse gap** behavior (`null` for empty or past-end slots vs 404 only for missing keys), **`encoding=buffer`** with null passthrough, **get-range** 1M span caps (including reversed ranges), **scan** `limit` validation and huge sparse spans without a span cap, and **ARNEXT** cursor semantics after `ARINSERT`/`ARSEEK`.
> 
> Each endpoint gets shared **validation**, **wrong-type / missing key / bad instance**, and **per-command ACL** cases. **Array create** tests gain cases for duplicate sparse indexes (last wins), the **2^64−1** reserved index, and empty sparse `elements`. **`keys/get-info`** array coverage moves to `test/api/array/POST-databases-id-keys-get_info-array.test.ts` so array-tagged RTE jobs run it without duplicating untagged suites.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 6704543126dd0731a8bfa2354bc06d20d6781aba. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: redisinsight/api/test/api/keys/POST-databases-id-keys-get_info.test.ts [hunk 1]
```diff
@@ -226,6 +226,10 @@ describe('POST /databases/:instanceId/keys/get-info', () => {
         },
       ].map(mainCheckFn);
     });
+
+    // Array coverage for this endpoint lives in
```
Comment (VaskoAtanasovRedis) [inline redisinsight/api/test/api/keys/POST-databases-id-keys-get_info.test.ts]: nit: not sure if this brings too much value as a comment

**Review by VaskoAtanasovRedis — DISMISSED** [review]

**Review by VaskoAtanasovRedis — APPROVED** [review]
